docs/winrt/WinRT SxS Components.md

```diff
@@ -1 +1,153 @@
+
+In this example, this is an update to Widgets which introduces non-breaking changes. This would be equivalent to semantic vesion 1.2
+
+```
+[component_version(1)]
+component Widgets {
+
+    [contract(Contoso.Widgets, 2)]
+    namespace Contoso.Widgets {
+
+        [contractversion(2.0)]
+        apicontract WidgetContract{};
+
+        [contract(Contoso.Widgets.WidgetContract, 1)]
+        runtime_class Doodad;
+
+        [contract(Contoso.Widgets.WidgetContract, 2)]
+        runtime_class Thingamabob;
+    }
+}
+```
+
+There is one non-obvious aspect to the example above. Notice that the contract is defined within the Widgets component. This means that the metadata name of the contract is actually Widgets_1.Contoso.Widgets.WidgetContract. MIDLRT will be updated to allow you to omit the contract "namespace" when referencing a contract defined within the same component. Although the names are similar under the covers, and look the same in MIDL3 text, Widgets_1.Contosos.Widgets.WidgetContract has no version relationship with Widgets_2.Contosos.Widgets.WidgetContract. They are two distinct and unrelated contracts.
+
+### Metadata Representation
+
+In the metadata, the MIDL above translates into a custom attribute component_vesion, as well as a namespace path element. To an unenlightened projection the metadata will appear as if the type Doodad contained an additional path element at the root. The MIDL 3 equivalent would be:
+
+```
+namespace Widgets_1.Contoso.Widgets {
+        runtime_class Doodad;
+}
+```
+
+Version 2 of the same component will end up with a different implicit namespace in metadata:
+
+```
+namespace Widgets_1.Contoso.Widgets {
+        runtime_class Doodad;
+}
+```
+
+### ABI & Platform support
+At the ABI, types will self-identify using the component prefix namespace. For example, the IInspectable::GetRuntimeClassName method for Doodad will return the string "Widgets_1.Contoso.Widgets.Doodad". This ensure that metadata resolution algorithms and metadata-based marshalling support in all versions of Windows that support it will be able to uniquely correlate runtime classes with metadata definitions, without risk of type confusion.
+
+The component-qualified type name will also be used in calculations for pinterfaces GUID calculations. This ensures that marshaling and dynamic casting via QueryInterface will not result in type confusion.
+
+Component authors that use explicit GUIDs for interfaces are encouraged, but not required to generate new GUIDs if the binary layout of an interface changes across major versions. 
+
+Type registrations will include the component namespace element in the registered type name. This ensures that activtion of different versions is unambiguous, and that an application can include more than one version of the component. Call to RoGetActivationFactory or RoActivateInstance should use the component-qualified name>
```

Comment:
~~activtion~~ -> activation

Reply:
There is a '>' at the end of the line. Is that intended?

Reply:
Fixed